src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,33 @@
+package seedu.addressbook.data.person;
+
+public abstract class Contact {
+    public final String value;
+    private boolean isPrivate;
+    
+    public Contact(String value, boolean isPrivate) {
+        this.value = value;
+        this.isPrivate = isPrivate;
+    }
+
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        Class<?> c = this.getClass();
+        return other == this // short circuit if same object
+                || (c.isInstance(other) // isInstance handles nulls
+                && this.value.equals(((Contact) other).value)); // state check
+    }
```

Comment:
This is fine, but note that now an Email and Address having the same value will be considered equal even if they are of different types.